File: src/main/java/page/JCB.java
package page;

import common.util.lang.LocaleCenter;

import javax.swing.*;

public class JCB extends JCheckBox implements LocComp {

    private static final long serialVersionUID = 1L;

    private final LocSubComp lsc;

    public JCB() {
        lsc = new LocSubComp(this);
        setBorder(BorderFactory.createEtchedBorder());
    }

    public JCB(LocaleCenter.Binder binder) {
        this();
        lsc.init(binder);
    }

    public JCB(int i, String str) {
        this();
        lsc.init(i, str);
    }

    public JCB(String str) {
        this(-1, str);
    }

    @Override
    public LocSubComp getLSC() {
        return lsc;
    }
}

File: src/main/java/page/JCB.kt
package page

import common.util.lang.LocaleCenter
import javax.swing.BorderFactory
import javax.swing.JCheckBox

class JCB() : JCheckBox(), LocComp {
    private val lsc: LocSubComp = LocSubComp(this)

    init {
        border = BorderFactory.createEtchedBorder()
    }

    constructor(binder: LocaleCenter.Binder) : this() {
        lsc.init(binder)
    }

    constructor(i: Int, str: String) : this() {
        lsc.init(i, str)
    }

    constructor(str: String) : this(-1, str)

    override fun getLSC(): LocSubComp {
        return lsc
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
